### ISSUE TITLE
Fix permissions on Docker

### ISSUE THREAD

**Issue by piranna** [NodeOS/NodeOS#331]
When running NodeOS on Docker, no matter what user do you login, it gets UID & GID 0, having root permissions. We need to find and fix this.

Docker AppArmor template
[disable filesystems mounting](https://github.com/docker/docker/blob/b89aff1afa1f61993ab2ba18fd62d9375a195f5d/profiles/apparmor/template.go#L31)
by default, so it's needed to disable usage of AppArmor at all. A better
alternative would be to use a custom template that mimics the Docker default one
but allowing filesystems mounting. This could be easily crafted with
[Bane](https://github.com/jessfraz/bane).

Docker default `seccomp` profile [accept setud](https://github.com/docker/docker/blob/475a8be5e507e3af645029336a75603e4fc1030d/profiles/seccomp/default.json#L303),
but for some reason it's not working and instead [they are a no-op](https://github.com/docker/docker/issues/31052)

**Comment by luii** [comment]
how do you check its uid and git inside nodeos?

**Comment by piranna** [comment]
Wth `pstree` and looking what processes tree it print, the one for `root` or the one of `nodeos` user. Dumb, but effective :-)

**Comment by luii** [comment]
what does your tree for root and for nodeos print?

**Comment by luii** [comment]
Im getting this:
```bash
~ > pstree
init
├── exclfs
└─┬ init
  ├─┬ nsh
  │ └── pstree
  └── nodeos-reverse-
```

**Comment by piranna** [comment]
> Im getting this (on docker):

This is the tree for `root` user.



> But it should be like this (on QEmu)?

Hum... that's not good, seems on QEmu we are having a regression, because it's too the list of `root` processes... In both cases they should be

```bash
~ > pstree
nsh
└ pstree
```

**Comment by luii** [comment]
This seem more like a software error from `logon` than from `docker` or `qemu`

**Comment by piranna** [comment]
> This seem more like a software error from logon than from docker or qemu

Docker has a problem, but it's true that `logon` was using `setreuid()` and changed it to `setuid()` because the docs says it's more secure. Maybe this could be the reason of the problem with QEmu... Problem here is about how to test it :-/

**Comment by luii** [comment]
You say that it was using `setreuid()` but the source on [here](https://github.com/piranna/logon/blob/master/server.js#L269-L270) is using `setreuid()` 
the usersfs pulls the dependencies on `build` so this could be the problem or am i wrong?

**Comment by piranna** [comment]
Yes, no matter what of them you use, they are from the same functions family. It's a problem on/with Docker.
